### PR TITLE
Feat stat share

### PR DESCRIPTION
**Pull request by willkoua**

### CODE REVIEW

File: src/test/java/com/wutsi/stats/impl/ClasspathInputStreamIteratorViewersTest.java [hunk 1]
```diff
@@ -8,7 +8,7 @@
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertNull;
 
-public class ClasspathInputStreamIteratorTest {
+public class ClasspathInputStreamIteratorViewersTest {
```
Comment (wutsi) [inline src/test/java/com/wutsi/stats/impl/ClasspathInputStreamIteratorViewersTest.java:11]: you should not rename this class
This iterator should always be the same for all

File: src/test/resources/tracks.shares/bot/2020-04-14-000.csv [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+"1586838312145","8c1413fd-2cba-47e1-aed0-f973402a4eaf","","page.read","scroll","2","25","Mac OS X","10","desktop","Chrome","","","","direct","","false","Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.80 Safari/537.36"
+"1586838408833","8c1413fd-2cba-47e1-aed0-f973402a4eaf","","page.read","scroll","2","30","Mac OS X","10","desktop","Chrome","","","","direct","","false","Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.80 Safari/537.36"
+"1586838413857","8c1413fd-2cba-47e1-aed0-f973402a4eaf","","page.read","share","2","linkedin","Mac OS X","10","desktop","Chrome","","","","direct","","false","Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.80 Safari/537.36"
+"1586838448079","8c1413fd-2cba-47e1-aed0-f973402a4eaf","","page.read","share","2","twitter","Mac OS X","10","desktop","Chrome","","","","direct","","false","Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.80 Safari/537.36"
```
Comment (wutsi) [inline src/test/resources/tracks.shares/bot/2020-04-14-000.csv:13]: this file has no share event associated with bot
Add one

File: src/test/resources/tracks.shares/happy-path/2020-04-14-001.csv [hunk 3]
```diff
@@ -0,0 +1,5 @@
+"time","deviceid","userid","page","event","productid","value","os","osversion","devicetype","browser","ip","long","lat","traffic","referer","bot","ua"
+"1586837219285","c8c2bd78-b0c4-4a7e-ae48-ca0942240e05","10","page.read","readstart","2","","Mac OS X","10","desktop","Chrome","","","","direct","","false","Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.80 Safari/537.36"
+"1586837221790","c8c2bd78-b0c4-4a7e-ae48-ca0942240e05","10","page.read","scroll","2","90","Mac OS X","10","desktop","Chrome","","","","direct","","false","Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.80 Safari/537.36"
+"1586837234672","c8c2bd78-b0c4-4a7e-ae48-ca0942240e05","10","page.read","scroll","2","100","Mac OS X","10","desktop","Chrome","","","","direct","","false","Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.80 Safari/537.36"
```
Comment (wutsi) [inline src/test/resources/tracks.shares/happy-path/2020-04-14-001.csv:4]: add a share event in this file
To confirm that you pick share events from all files
- Add a share event for productid=2
- add another share event for productid=4

File: src/main/java/com/wutsi/stats/shares/DailySharesAggregator.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        String trackCurrentDate = this.convertTimeStampToDateString(Long.parseLong(trackDate));
+        return this.date.toString().equals(trackCurrentDate);
+    }
+
+    private String convertTimeStampToDateString (long timestamp) {
+        DateFormat df = new SimpleDateFormat("yyyy-MM-dd");
+        return df.format(new Date(timestamp).getTime());
+    }
+
+    private List<Share> getViewers(InputStreamIterator iterator) throws IOException {
```
Comment (wutsi) [inline src/main/java/com/wutsi/stats/shares/DailySharesAggregator.java:60]: rename ``getShares()``

File: src/main/java/com/wutsi/stats/shares/DailySharesAggregator.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        return getViewers(this.getTracks(iterator));
+    }
+
+    private List<Share> getViewers(List<Track> tracks) {
```
Comment (wutsi) [inline src/main/java/com/wutsi/stats/shares/DailySharesAggregator.java:64]: rename ``getShares()``

File: src/test/resources/tracks.shares/bot/2020-04-14-000.csv (hunk 2 above)
Comment (willkoua) [reply src/test/resources/tracks.shares/bot/2020-04-14-000.csv:13]: done

File: src/test/resources/tracks.shares/happy-path/2020-04-14-001.csv (hunk 3 above)
Comment (willkoua) [reply src/test/resources/tracks.shares/happy-path/2020-04-14-001.csv:4]: done

**Pull request merged**